[Regression] justine.lol: Video element blank showing no poster or thumbnail

jyavenard opened the pull request:
#### ab70b0f28767c28a114b9a6a7d075937dcec813f
<pre>
[Regression] justine.lol: Video element blank showing no poster or thumbnail
<a href="https://bugs.webkit.org/show_bug.cgi?id=291462">https://bugs.webkit.org/show_bug.cgi?id=291462</a>
<a href="https://rdar.apple.com/146583593">rdar://146583593</a>

Reviewed by Eric Carlson.

When partial mkv support was added 289843@main we stopped treating as error
invalid track type (h264 and PCM).
If allowLimitedMatroska is off (default) treat h264 and pcm tracks as error.

Add tests.
* LayoutTests/media/content/test-h264.mkv: Added.
* LayoutTests/media/media-h264-webm-error-expected.txt: Added.
* LayoutTests/media/media-h264-webm-error.html: Added.
* LayoutTests/media/media-h264-webm-source-expected.txt: Added.
* LayoutTests/media/media-h264-webm-source.html: Added.
* LayoutTests/platform/glib/media/media-h264-webm-error-expected.txt: Added. GStreamer webm player allows for mkv to play.
* LayoutTests/platform/glib/media/media-h264-webm-source-expected.txt: Added. Ditto
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::isSupportedVideoCodec):
(WebCore::WebMParser::isSupportedAudioCodec):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.h:

Canonical link: <a href="https://commits.webkit.org/293641@main">https://commits.webkit.org/293641@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/39a24b1ecdd0d4daa3d7fab2d2221a9fe48475eb

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/99454 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/19104 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/9357 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/104585 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/50055 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/101495 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/19392 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/27537 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75695 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32795 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/102461 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14759 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89801 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/56054 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/14555 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/7784 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/49415 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/84477 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7872 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106943 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/26568 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/19383 "Found 1 new test failure: editing/execCommand/apply-inline-style-to-element-with-no-renderer-crash.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84654 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26930 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/86005 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/84172 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21362 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28846 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/6541 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/20327 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/26508 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/31709 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/26328 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/29641 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27895 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

eric-carlson commented on LayoutTests/media/media-h264-webm-error.html:
```diff
@@ -0,0 +1,35 @@
+
+            video.onerror = (event) => {
+                consoleWrite("received error event");
+                endTest();
```
Nit: maybe `failTest` instead of `endTest`?

Ditto for the failure cases in the other tests.

jyavenard replied:
> Nit: maybe `failTest` instead of `endTest`?
> 
> Ditto for the failure cases in the other tests.

I use the different output between gstreamer player and AVF.
WPE and GTK will play those videos just fine (which is new). AVF doesn't.

So I set different expectations between platforms.
emitting an error with AVFObjC players is the expected outcome.

jyavenard committed:
[Regression] justine.lol: Video element blank showing no poster or thumbnail
https://bugs.webkit.org/show_bug.cgi?id=291462
rdar://146583593

Reviewed by Eric Carlson.

When partial mkv support was added 289843@main we stopped treating as error
invalid track type (h264 and PCM).
If allowLimitedMatroska is off (default) treat h264 and pcm tracks as error.

Add tests.
* LayoutTests/media/content/test-h264.mkv: Added.
* LayoutTests/media/media-h264-webm-error-expected.txt: Added.
* LayoutTests/media/media-h264-webm-error.html: Added.
* LayoutTests/media/media-h264-webm-source-expected.txt: Added.
* LayoutTests/media/media-h264-webm-source.html: Added.
* LayoutTests/platform/glib/media/media-h264-webm-error-expected.txt: Added. GStreamer webm player allows for mkv to play.
* LayoutTests/platform/glib/media/media-h264-webm-source-expected.txt: Added. Ditto
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.cpp:
(WebCore::WebMParser::isSupportedVideoCodec):
(WebCore::WebMParser::isSupportedAudioCodec):
* Source/WebCore/platform/graphics/cocoa/SourceBufferParserWebM.h:

Canonical link: https://commits.webkit.org/293641@main
